android/src/main/kotlin/com/kasem/media_picker_builder/providers/VideoFileProvider.kt

```diff
@@ -2,93 +2,189 @@ package com.kasem.media_picker_builder.providers
+                            MediaStore.Video.Media.DATE_ADDED,
+                            MediaStore.Video.Media.DATA,
+                            MediaStore.Video.Media.MIME_TYPE,
+                            MediaStore.Video.Media.BUCKET_ID,
+                            MediaStore.Video.Media.DURATION,
+                            MediaStore.Video.Media.ORIENTATION),
+                    selectionClause,
+                    selectionArgs,
+                    "${MediaStore.Video.Media._ID} DESC")
+
+            while (cursor?.moveToNext() == true) {
+                val mediaFile = cursorToMediaFileSubsequent29(cursor)
+
+                if (mediaFile != null) {
+                    val album = albumHashMap[mediaFile.albumId]
+                    if (album == null) {
+                        albumHashMap[mediaFile.albumId] = Album(
+                                mediaFile.albumId,
+                                mediaFile.albumName,
+                                mutableListOf(mediaFile)
+                        )
+                    } else {
+                        album.files.add(mediaFile)
+                    }
+                }
+            }
+        } else {
+            val cursor = context.contentResolver.query(
+                    MediaStore.Video.Media.EXTERNAL_CONTENT_URI,
+                    arrayOf(
+                            MediaStore.Video.Media._ID,
+                            MediaStore.Video.Media.DATE_ADDED,
+                            MediaStore.Video.Media.DATA,
+                            MediaStore.Video.Media.MIME_TYPE),
+                    selectionClause,
+                    selectionArgs,
+                    "${MediaStore.Video.Media._ID} DESC")
+
+            val mediaMetadataRetriever = MediaMetadataRetriever()
+            while (cursor?.moveToNext() == true) {
+                val mediaFile = cursorToMediaFilePriorTo29(mediaMetadataRetriever, cursor)
+
+                if (mediaFile != null) {
+                    val album = albumHashMap[mediaFile.albumId]
+                    if (album == null) {
+                        albumHashMap[mediaFile.albumId] = Album(
+                                mediaFile.albumId,
+                                mediaFile.albumName,
+                                mutableListOf(mediaFile)
+                        )
+                    } else {
+                        album.files.add(mediaFile)
+                    }
                 }
             }
+            mediaMetadataRetriever.release()
         }
     }
 
-    private fun cursorToMediaFile(cursor: Cursor): MediaFile {
-        val fileId = cursor.getLong(0)          //MediaStore.Video.Media._ID
-        val fileDateAdded = cursor.getLong(1)   //MediaStore.Video.Media.DATE_ADDED
-        val filePath = cursor.getString(2)      //MediaStore.Video.Media.DATA
-        val albumName = cursor.getString(3)     //MediaStore.Video.Media.BUCKET_DISPLAY_NAME
```

Comment:
Why is this removed from the new implementation?

Reply:
albumName seems to have some problems that was commented. Also we don't use it

Reply:
albumName fixed